### PR TITLE
[Event Hubs] August Release Prep

### PR DESCRIPTION
**Pull request by jsquire**
# Summary

The focus of these changes is to prepare the core Event Hubs package for the August release.

### CODE REVIEW

File: sdk/eventhub/Azure.Messaging.EventHubs/CHANGELOG.md [hunk 1]
```diff
@@ -1,15 +1,11 @@
 # Release History
 
-## 5.6.0-beta.1 (Unreleased)
+## 5.5.1 (2021-06-09)
```
Comment (JoshLove-msft) [inline sdk/eventhub/Azure.Messaging.EventHubs/CHANGELOG.md:3]:
```suggestion
## 5.5.1 (2021-08-09)
```

Comment (JoshLove-msft) [inline sdk/eventhub/Azure.Messaging.EventHubs/CHANGELOG.md:3]: Should this be a minor version bump since we are changing the API surface?

Comment (jsquire) [reply sdk/eventhub/Azure.Messaging.EventHubs/CHANGELOG.md:3]: Good point.   It probably should.

**Review by JoshLove-msft — APPROVED** [review]

**Pull request merged**